tensorflow/core/kernels/sparse_reorder_op.cc

```diff
@@ -54,9 +55,10 @@ class SparseReorderOp : public OpKernel {
                     "Input shape should be a vector but received shape ",
                     input_shape_in.shape().DebugString()));
 
-    const TensorShape input_shape(input_shape_in.vec<int64>());
+    gtl::ArraySlice<int64> input_shape(
+        input_shape_in.vec<int64>().data(), input_shape_in.NumElements());
 
-    gtl::InlinedVector<int64, 8> std_order(input_shape.dims());
+    gtl::InlinedVector<int64, 8> std_order(input_shape.begin(), input_shape.end());
```

Comment:
Let's add comments here for why we are using the iterator based versions instead of the constructors that take a single argument.

Reply:
Thanks @mihaimaruseac. I checked again, as `std_order` is actually initialized through the following line `std::iota(...)`, `std_order` only needs to be initialized with appropriate size (no need to get the copy of the shape). For that I have replace the initialization of `std_order` with `gtl::InlinedVector<int64, 8> std_order(input_shape.size())`.